src/graphql/resolveProcedure.js

```diff
@@ -34,13 +39,26 @@ const resolveProcedure = (procedure, getProcedureArgs) => {
   return async (source, args, { client }) => {
     const procedureArgs = getProcedureArgs(source, args)
 
-    // Actuall run the procedure using our arguments.
-    const result = await client.queryAsync({
-      name: query.name,
-      text: query.text,
-      values: argEntries.map(([name]) => procedureArgs[camelCase(name)]),
+    const values = argEntries.map(([name, type]) => {
+      const obj = procedureArgs[camelCase(name)]
+      let value
+
+      if (type.isTableType) {
+        value = reduce(obj, (acc, val, key) => {
```

Comment:
It should be fine for now then. Just some small changes, instead of `reduce` could you use `mapValues`? Also could you rename `convertToPostgres`… to `asPgValue`. It's shorter and follows our `pg` convention. Also it can be extended to support more cases in the future 😊